src/library.js

```diff
@@ -1655,6 +1655,16 @@ addToLibrary({
   emscripten_asm_const_int_sync_on_main_thread__deps: ['$runMainThreadEmAsm'],
   emscripten_asm_const_int_sync_on_main_thread: (emAsmAddr, sigPtr, argbuf) => runMainThreadEmAsm(emAsmAddr, sigPtr, argbuf, 1),
 
+  emscripten_asm_const_int_await_promise_on_main_thread__deps: ['$runMainThreadEmAsm'],
+  emscripten_asm_const_int_await_promise_on_main_thread: (emAsmAddr, sigPtr, argbuf) => {
```

Comment:
How about we drop the `promise` part so this becomes `emscripten_asm_const_int_await_on_main_thread` and the macro is `MAIN_THREAD_EM_ASM_AWAIT`?

Reply:
Yes, I like that better as well